core/domain/collection_domain.py

```diff
@@ -1064,16 +1070,40 @@ def delete_node(self, exploration_id):
         del self.nodes[node_index]
 
     def add_skill(self, skill_name):
-        """Adds the new skill domain object with the specified name."""
+        """Adds the new skill domain object with the specified name.
+
+        Args:
+            skill_name: str. The name of the skill.
 
+        Returns
+            str. The id of the new skill.
+        """
         for _, skill in self.skills.iteritems():
```

Comment:
for skill in self.skills.itervalues()

or even

`if any([skill_name == skill.name for skill in self.skills.itervalues()])`

Reply:
Updated.